src/library-authoring/components/ComponentDeleter.tsx

```diff
@@ -1,48 +1,37 @@
 import React, { useCallback, useContext } from 'react';
 import { FormattedMessage, useIntl } from '@edx/frontend-platform/i18n';
 import { Icon } from '@openedx/paragon';
-import { School, Warning, Info } from '@openedx/paragon/icons';
+import { School, Delete, Info } from '@openedx/paragon/icons';
 
+import { useEntityLinks } from '@src/course-libraries/data/apiHooks';
+import DeleteModal from '@src/generic/delete-modal/DeleteModal';
+import { ToastContext } from '@src/generic/toast-context';
+import { type ContentHit } from '@src/search-manager';
 import { useSidebarContext } from '../common/context/SidebarContext';
+import { useLibraryContext } from '../common/context/LibraryContext';
 import {
   useContentFromSearchIndex,
   useDeleteLibraryBlock,
-  useLibraryBlockMetadata,
   useRestoreLibraryBlock,
 } from '../data/apiHooks';
 import messages from './messages';
-import { ToastContext } from '../../generic/toast-context';
-import DeleteModal from '../../generic/delete-modal/DeleteModal';
-import { type ContentHit } from '../../search-manager';
-
-/**
- * Helper component to load and display the name of the block.
- *
- * This needs to be a separate component so that we only query the metadata of
- * the block when needed (when this is displayed), not on every card shown in
- * the search results.
- */
-const BlockName = (props: { usageKey: string }) => {
-  const { data: blockMetadata } = useLibraryBlockMetadata(props.usageKey);
-
-  // eslint-disable-next-line react/jsx-no-useless-fragment
-  return <>{blockMetadata?.displayName}</> ?? <FormattedMessage {...messages.deleteComponentNamePlaceholder} />;
-};
```

Comment:
Now we only render the Delete/Remove dialog if the `isOpen` is visible, so we don't need this.